### ISSUE TITLE
Lightweight Logging System

### ISSUE THREAD

**Issue by Manuzor** [lab132/PyBake#9]
I think it would be good to have a light-weight logging system that takes care of logging messages according to the current verbosity level.

A feature I would like to see would be the ability to structure log messages, just like with `EZ_LOG_BLOCK`.

**Comment by pampersrocker** [comment]
Any idea how this structuring could work? RAII with scopes is not possible in python so you would need to write something like

```python
def myLoggedFunction():
    beginLog("MyFunctionScope")
    # .....
    endLog()
```

This is certainly a way to go but might cause issues because you forgot to put `endLog` there and your logging is not being printed.

Maybe a more convenient way would be to define log blocks per function using a decorator:

```python
@logged
def myLoggedFunction():
    #....
    log(VeryVerbose,"Spam")
```

where the stuff from the first example would be in the logged decorator like

```python
def logged(name=None):
    beginLogBlock(name)
    func()
    endLogBlock()
```

**Comment by Manuzor** [comment]
Check out `ChangeDir` in `__init__.py`. It's used like this:

~~~{.py}
with ChangeDir(".."):
  # Do stuff in ..
# We are in . again
~~~

So it could look like:

~~~{.py}
with LogBlock("Stuff happens here")
  with LogBlock("Deeply nested stuff...")
    logMessage("woot, woot!")

# Output:
+++ Stuff happens here
  +++ Deeply nested stuff... 
    woot, woot!
  --- Deeply nested stuff...
--- Stuff happens here
~~~

Sure, you'd have to indent everytime you wanted such a block, but:
1. You could create a log-block manually and call `enter`/`exit` without extra indentation
1. This structures the code in quite a natural way

**Comment by Manuzor** [comment]
Note: `EZ_LOG_BLOCK` only logs the log-block message, if any logging happens within the active block, i.e. if you enter a log-block but do not log any message in there, the log-blocks will not write their headings. This is useful in case you only want to log errors.

**Comment by pampersrocker** [comment]
Basic logging facility was implemented in 16d19b1

**Comment by pampersrocker** [comment]
Was implemented and merged in #11
